lib/dest/write-contents/write-buffer.js

```diff
@@ -1,14 +1,30 @@
 'use strict';
 
+var iconv = require('iconv-lite');
+
 var fo = require('../../file-operations');
+var getCodec = require('../../codecs');
 
 function writeBuffer(file, optResolver, onWritten) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
+  if (encoding && !codec) {
+    return onWritten(new Error('Unsupported encoding: ' + encoding));
+  }
+
   var opt = {
     mode: file.stat.mode,
     flag: optResolver.resolve('flag', file),
   };
 
-  fo.writeFile(file.path, file.contents, opt, onWriteFile);
+  var contents = file.contents;
+
+  if (encoding && codec.enc !== 'utf8') {
+    contents = iconv.decode(contents, 'utf8', { stripBOM: false });
```

Comment:
If you decode to utf8 then encode as a different encoding, won't characters that utf8 doesn't support be changed/broken/something?